### PR TITLE
updating assistant

### PR DESCRIPTION
**Pull request by haraldschilly**
* this updates the submodule, you can see all the changes [here](https://github.com/sagemathinc/cocalc-assistant/compare/6e2ca707d2220d04d3651ec339f2117267813639...805f5699e16698a649a3929a2e8c83401edad0d0). in particular, it adds gap examples, fixes many sage issues, and adds the whole test infrastructure.
* the assistant itself inserts variables if they're known, such that a couple of problematic cases are no longer triggered. this is shown in the screenshot below.

### CODE REVIEW

**Comment by haraldschilly** [comment]
@tscholl2  I made some additional to the data regarding initialization variables. this code here is the second part, which adds relevant lines to the setup code. here is a screenshot as an example for one of these special functions:

screenshot from 2018-08-07 17-59-34

screenshot from 2018-08-07 17-58-53

**Comment by williamstein** [comment]
I tried to referee this but didn't even know what to do.

**Comment by haraldschilly** [comment]
To checkout this at all, do

1. stop webpack watcher
2. switch branch
3. start webpack watcher again (it should do `git submodule update` and rebuild the assistant data file)

Besides that, I've merged this branch with master again. I did a couple of spot checks and looks fine to me. I don't know how you want to referee this. The top comment has a link to all changed examples.

Actual changes relevant to the cocalc codebase:

**predefined variables** which is visible in the two screenshots in an earlier comment above: 

1. make a notebook, python3 ubuntu linux kernel
2. type in `order 0` into the search box and select the second one: "j0: Bessel function the first kind of order 0, J_0..." and select it
3. toggle the "setup code" button at the bottom, to see that there is an additional `x = 1.5`. Hence when it runs (inserted when setup code is selected!), x will be defined.

**multicell inserts** for GAP

1. make a gap notebook
2. see that there is no longer nothing for gap
3. select introduction → basics → list. after inserting, you'll have several list-related cells, not just one

... but don't bother testing it, it's a crying shame with gap :cry: 

screenshot from 2018-09-06 19-08-10

File: src/smc-webapp/assistant/actions.coffee [hunk 1]
```diff
@@ -177,12 +182,54 @@ class exports.ExamplesActions extends Actions
...
+            # assuming function calls are after the first open ( bracket
+            re = /\b([a-zA-Z_0-9]+)/g
+            # all detected variable names are collected in that array
+            varincode = []
+            for line in code.split('\n')
+                if '(' in line
+                    line = line[line.indexOf('(')...]
+                line.replace(re, ((_, g) -> varincode.push(g)))
+            # then we add name = values lines to set only these
+            # TODO syntax needs to be language specific!
```
Comment (williamstein) [inline src/smc-webapp/assistant/actions.coffee]: Please make a ticket for this TODO!

**Pull request merged**